frontends/p4/tableKeyNames.cpp

```diff
@@ -32,10 +38,16 @@ class KeyNameGenerator : public Inspector {
 
     void postorder(const IR::Expression* expression) override { error(expression); }
 
+    /** Compute a name annotation for @expression.  Eg. `@name("foo.bar")` for
```

Comment:
Sorry, I read "These comments" more broadly and misunderstood. 

I agree that too many comments can be overkill. I would be in favor of class comments that document the assumptions / guarantees of each pass and the overall strategy used to implement it. Members of the class and helper functions may also merit comments but we should use our judgment. I agree that commenting each pre/post-order method may be overkill if it's just implementing the functionality specified in the class comment, unless there is something exceptional going on in that method.